runtime/common/src/main/kotlin/kotlinx/serialization/internal/CollectionSerializers.kt

```diff
@@ -37,9 +39,7 @@ sealed class ListLikeSerializer<E, C, B>(open val eSerializer: KSerializer<E>) :
     abstract fun B.checkCapacity(size: Int)
     abstract fun B.insert(index: Int, element: E)
 
-    open val typeParams: Array<KSerializer<*>> = arrayOf(eSerializer)
```

Comment:
There was no need for this to be open. Instead, this is now passed through the constructor.